projects/ui/autocomplete/autocomplete-trigger.ts

```diff
@@ -0,0 +1,720 @@
+import { Directionality } from '@angular/cdk/bidi'
+import {
+  DOWN_ARROW,
+  ENTER,
+  ESCAPE,
+  hasModifierKey,
+  TAB,
+  UP_ARROW,
+} from '@angular/cdk/keycodes'
+import {
+  ConnectedPosition,
+  FlexibleConnectedPositionStrategy,
+  Overlay,
+  OverlayConfig,
+  OverlayRef,
+  ScrollStrategy,
+} from '@angular/cdk/overlay'
+import { _getEventTarget } from '@angular/cdk/platform'
+import { TemplatePortal } from '@angular/cdk/portal'
+import { DOCUMENT } from '@angular/common'
+import {
+  afterNextRender,
+  booleanAttribute,
+  ChangeDetectorRef,
+  computed,
+  Directive,
+  effect,
+  ElementRef,
+  forwardRef,
+  Host,
+  Inject,
+  Injector,
+  input,
+  NgZone,
+  numberAttribute,
+  OnDestroy,
+  Optional,
+  signal,
+  untracked,
+  ViewContainerRef,
+} from '@angular/core'
+import {
+  outputToObservable,
+  toObservable,
+  toSignal,
+} from '@angular/core/rxjs-interop'
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms'
+import { CkFormField } from '@corekit/ui/form-field'
+import { CkOption } from '@corekit/ui/option'
+import {
+  filter,
+  firstValueFrom,
+  fromEvent,
+  map,
+  merge,
+  Observable,
+  startWith,
+  switchMap,
+  take,
+} from 'rxjs'
+import { CkAutocomplete } from './autocomplete'
+import { CkAutocompleteOrigin } from './autocomplete-origin'
+import { CK_AUTOCOMPLETE_SCROLL_STRATEGY } from './scroll-strategy.token'
+
+type AutocompletePosition = 'auto' | 'above' | 'below'
+
+const CONTROL_VALUE_ACCESSOR_PROVIDER = {
+  provide: NG_VALUE_ACCESSOR,
+  useExisting: forwardRef(() => CkAutocompleteTrigger),
+  multi: true,
+}
+
+/** Directive that adds autocomplete feature to an input or a textarea. */
+@Directive({
+  selector:
+    'input[ck-autocomplete-trigger-for], textarea[ck-autocomplete-trigger-for]',
```

Comment:
Maybe will make it easier? Just `input[ckAutocomplete]`

Reply:
Yea, also was doing that while I was developing, but gave up in the end and went with the approach Material uses. For me it doesn't make a lot of difference, but one of the ideas of this lib was to make things as close to the Material API as possible, introducing changes only if that makes things easier.

And it also makes perfect sense if you look at it as at plain English sentence:

```html
<input [ck-autocomplete-trigger-for]="autocomplete">
```
which for me reads like "This is an input, and it is an autocomplete trigger for this specific {autocomplete}"